pkg/cache/task.go

```diff
@@ -543,7 +565,62 @@ func (task *Task) releaseAllocation() {
 // some sanity checks before sending task for scheduling,
 // this reduces the scheduling overhead by blocking such
 // request away from the core scheduler.
-func (task *Task) sanityCheckBeforeScheduling() error {
+func (task *Task) sanityCheckBeforeScheduling() (error, bool) {
+	rejectTask := false
```

Comment:
it seems `rejectTask` is always "false"?

Reply:
Yes, rejectTask is always "false" for 1.6.0.
In 1.7.0, the rejectTask will be "true" if the sanity check fail due to inconsistent matadata.

Reply:
Sorry for the unclear comment, I just added the hardcode part in the latest commit.
If the sanity check fail due to inconsistent metadata, the sanity check should return err and set the `rejectTask` to true. (Only happen after in 1.7.0)


https://github.com/chenyulin0719/yunikorn-k8shim/blob/7f8b8652953cda033a51da17fa1b0a222f24eb8c/pkg/cache/task.go#L587-L589